components/automate-ui/src/app/pages/create-notification-modal/create-notification-modal.component.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit
+} from '@angular/core';
+import { combineLatest, Subject } from 'rxjs';
+import { Store, select } from '@ngrx/store';
+import { FormBuilder,  Validators, FormGroup } from '@angular/forms';
+import { filter, takeUntil } from 'rxjs/operators';
+import { isNil } from 'lodash/fp';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { Regex } from 'app/helpers/auth/regex';
+import { EntityStatus, pending } from 'app/entities/entities';
+import { HttpStatus } from 'app/types/types';
+import {
+  saveStatus,
+  saveError
+} from 'app/entities/notification_rules/notification_rule.selectors';
+import {
+  NotificationRule,
+  ServiceActionType,
+  RuleType
+} from 'app/entities/notification_rules/notification_rule.model';
+import {
+  CreateNotificationRule
+} from 'app/entities/notification_rules/notification_rule.action';
+import {
+  NotificationRuleRequests
+} from 'app/entities/notification_rules/notification_rule.requests';
+
+enum UrlTestState {
+  Inactive,
+  Loading,
+  Success,
+  Failure
+}
+
+@Component({
+  selector: 'app-create-notification-modal',
+  templateUrl: './create-notification-modal.component.html',
+  styleUrls: ['./create-notification-modal.component.scss']
+})
+export class CreateNotificationModalComponent implements OnInit {
+  @Input() openEvent: EventEmitter<boolean>;
+
+  public visible = false;
+  public creating = false;
+  public sending = false;
+  public conflictErrorEvent = false;
+  public close = new EventEmitter();
+  public createForm: FormGroup;
+  public hookStatus = UrlTestState.Inactive;
+  public notificationRule = new NotificationRule('', '', null, '', null, '', false);
+  public conflictError = false;
+  public urlState = UrlTestState;
+  public targetKeys: string[];
+  public alertTypeKeys: string[];
+  private isDestroyed = new Subject<boolean>();
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    private fb: FormBuilder,
+    private notificationRuleRequests: NotificationRuleRequests
+  ) {
+    this.createForm = this.fb.group({
+      // Must stay in sync with error checks in create-notification-modal.component.html
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      ruleType: ['', [Validators.required]],
+      targetType: ['', [Validators.required]],
+
+      // Note that URL here may be FQDN -or- IP!
+      targetUrl: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      username: [[]],
+      password: [[]]
+    });
+
+    this.targetKeys = this.notificationRule.getTargetTypeKeys();
+    this.alertTypeKeys = this.notificationRule.getAlertTypeKeys();
+  }
+
+  ngOnInit() {
+    this.openEvent.pipe(takeUntil(this.isDestroyed))
```

Comment:
You have everything nicely implemented with the `takeUntil` -- but `this.isDestroyed` never fires because you do not implement `OnDestroy` in the class.